src/main/java/com/ldtteam/structurize/config/BlueprintRenderSettings.java

```diff
@@ -47,21 +68,147 @@ public BlueprintRenderSettings(final FriendlyByteBuf byteBuf)
+            // client instance
+            key.forgeConfig.get() :
+            // server instance (or client instance from buffer)
+            (T) renderSettings.get(key.id);
+    }
+
+    public <T> void setSetting(final SharedSettings<T> key, final T value)
+    {
+        if (renderSettings == null)
+        {
+            key.forgeConfig.set(value);
+            Structurize.getConfig().onConfigValueEdit(key.forgeConfig);
+        }
+        else
+        {
+            renderSettings.set(key.id, value);
+        }
+    }
+
+    /**
+     * @return list of client config values via their origin (config or shared settings)
+     */
+    public static List<EitherConfig<?>> gatherClientRendererConfigs()
+    {
+        final List<EitherConfig<?>> result = new ArrayList<>();
+        final Set<ConfigValue<?>> deduplicator = Collections.newSetFromMap(new IdentityHashMap<>());
+
+        // add shared first so the gui knows the proper origin
+        SharedSettings.VALUES.forEach(value -> {
+            if (deduplicator.add(value.forgeConfig))
+            {
+                result.add(EitherConfig.shared(value));
+            }
+        });
+        Structurize.getConfig().getClient().collectPreviewRendererSettings(value -> {
+            if (deduplicator.add(value))
+            {
+                result.add(EitherConfig.normal(value));
+            }
+        });
+
+        return result;
+    }
+
+    /**
+     * Enum-like record for settings which should be shared through preview sharing
+     */
+    public record SharedSettings<T>(int id,
+        ConfigValue<T> forgeConfig,
+        BiConsumer<FriendlyByteBuf, T> bufferWriter,
+        Function<FriendlyByteBuf, T> bufferReader)
+    {
+        public static final List<SharedSettings<?>> VALUES = new ArrayList<>();
+
+        private SharedSettings(final ConfigValue<T> forgeConfig,
+            final BiConsumer<FriendlyByteBuf, T> bufferWriter,
+            final Function<FriendlyByteBuf, T> bufferReader)
+        {
+            // id from size ensures that #renderSettings have same order
+            this(VALUES.size(), forgeConfig, bufferWriter, bufferReader);
+            VALUES.add(this);
+        }
+
+        public void writeClientToBuf(final FriendlyByteBuf buffer)
+        {
+            bufferWriter.accept(buffer, forgeConfig.get());
+        }
+
+        @SuppressWarnings("unchecked")
+        public void writeToBuf(final FriendlyByteBuf buffer, final Object obj)
         {
-            byteBuf.writeUtf(entry.getKey());
-            byteBuf.writeBoolean(entry.getValue());
+            bufferWriter.accept(buffer, (T) obj);
         }
     }
 
     /**
-     * Register a new setting.
-     * @param key the settings key.
-     * @param value the settings value.
+     * Wrapper to merge normal and shared configs
      */
-    public void registerSetting(final String key, final boolean value)
+    public record EitherConfig<T>(ConfigValue<T> normal, SharedSettings<T> wrapped)
```

Comment:
Im not sure I get what all of this boilerplate here is for?

Reply:
You made it, I just kept functionality, u tell me :P
Like for now it's just one boolean, but this allows any thing to be sent through

Reply:
I mean the ton of new settings and records and stuff here =D

Reply:
to keep ur functionality and merge with normal configvalues from normal configs, i was also thinking about reducing to just one boolean (cuz it was one boolean in link-sessions anyway), but u made it like this so i kept it logic-wise same, just binary incompatible cuz normal configvalues

Reply:
Do we use those in minecolonies? I mean, I would make this now just "only client configs" and get rid of the old one.

Reply:
Entire class deleted based on discord convo